app/lib/services/domain_ownership_service.rb

```diff
@@ -1,14 +1,53 @@
 module Services
   class DomainOwnershipService
-    def authorize!(identity, cert_req)
-      cert_req.fqdns.each do |fqdn|
-        domain = Domain.where(fqdn: fqdn).first
-        raise AuthError unless domain.present? &&
-                               (domain.owner == identity.subject ||
-                               (domain.group_delegation &&
-                               (domain.groups & identity.groups).any?))
+    attr_reader :client
+
+    def initialize
+      @client = Faraday.new(ssl: ssl_opts, url: Rails.configuration.astral[:app_registry_addr]) do |faraday|
+        faraday.request :authorization, "Bearer", -> { Rails.configuration.astral[:app_registry_token] }
+        faraday.request :retry, retry_opts
+        faraday.response :json
+        faraday.response :raise_error, include_request: true
       end
+    end
+
+    def get_domain_info(fqdn)
```

Comment:
Should we follow a pattern of making these class methods instead of instance methods? The calling code looked weird to me.

Reply:
It seems more Railsy to do it the singleton way, ie User.first etc etc

Reply:
converted these to class-method interfaces